tree-sitter/src/TreeSitter/Tree.hs

```diff
@@ -16,5 +20,35 @@ withRootNode tree action = alloca $ \ ptr -> do
   ts_tree_root_node_p tree ptr
   action ptr
 
+data TSInputEdit = TSInputEdit
```

Comment:
Little nitpick: Could you please add some form of documentation to this struct? Does it reflect something from the C library? If yes, could you perhaps point tot he